cmd/cloud_sql_proxy/internal/healthcheck/healthcheck.go

```diff
@@ -62,6 +63,15 @@ func NewServer(c *proxy.Client, port string) (*Server, error) {
 		srv:  srv,
 	}
 
+	mux.HandleFunc(startupPath, func(w http.ResponseWriter, _ *http.Request) {
+		if !hcServer.proxyStarted() {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			w.Write([]byte("error"))
```

Comment:
You'll need a `return` here. Otherwise, the following writes will also be written to the ResponseWriter.